File: src/main/java/net/miarma/mkernel/commands/roleplay/DoCommand.java
package net.miarma.mkernel.commands.roleplay;

import java.util.Arrays;
import java.util.stream.Collectors;

import org.bukkit.Bukkit;

import dev.jorel.commandapi.CommandAPICommand;
import net.miarma.mkernel.config.CommandWrapper;
import net.miarma.mkernel.config.providers.CommandProvider;
import net.miarma.mkernel.util.PlayerUtil;

public class DoCommand {
    public static void register() {
        CommandWrapper doCmd = CommandProvider.getDoCommand();
        new CommandAPICommand(doCmd.getName())
            .withArguments(CommandProvider.Arguments.message())
            .withFullDescription(doCmd.getDescription())
            .withPermission(doCmd.getPermission().base())
            .withShortDescription(doCmd.getDescription())
            .executesPlayer((sender, args) -> {
                String joinedArgs = Arrays.stream(args.rawArgs()).collect(Collectors.joining(" "));
                String msg = "§9(" + sender.getName() + ") [Do] §7" + joinedArgs;
                Bukkit.getServer().getOnlinePlayers().stream()
                        .filter(p -> (p.getWorld() == sender.getWorld()) && (PlayerUtil.distance(sender, p) < 25 || sender.equals(p)))
                        .forEach(p -> p.sendMessage(msg));
            })
            .register();
    }
}

File: src/main/kotlin/net/miarma/mkernel/command/impl/roleplay/DoCommand.kt
package net.miarma.mkernel.command.impl.roleplay

import com.google.inject.Inject
import com.google.inject.Singleton
import dev.jorel.commandapi.arguments.GreedyStringArgument
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.playerExecutor
import net.miarma.mkernel.command.MCommand
import net.miarma.mkernel.common.service.impl.ConfigService
import net.miarma.mkernel.common.service.impl.MessageService
import org.bukkit.Bukkit

@Singleton
class DoCommand @Inject constructor(
    private val configService: ConfigService,
    private val messageService: MessageService
) : MCommand {
    override fun register() {
        commandAPICommand(configService.getString("commands.do.name")) {
            withArguments(GreedyStringArgument(configService.getString("arguments.message")))
            withFullDescription(configService.getString("commands.do.description"))
            withPermission(configService.getString("commands.do.permission"))
            playerExecutor { sender, args ->
                val message = args[0] as String
                val builder = messageService.builder("<blue>(<player>) [Do] <gray><message>")
                    .tag("player", sender.name)
                    .tag("message", message)

                Bukkit.getServer().onlinePlayers
                    .filter { p -> p.world == sender.world && p.location.distance(sender.location) < 25 }
                    .forEach { p -> builder.send(p) }
            }
        }
    }
}